    bool out_flag
    int32 width
    int32 height
    int32 x_pos
    int32 y_pos